File: Singleton01.py
class Singleton:
    instance = None 
    def __new__(cls):
        if cls.instance is None:
            cls.instance = super().__new__(cls)
            cls.instance.value = 0
        return cls.instance
    
    def increment(self):
        self.value += 1
        print(f"Value : {self.value}")

######## CLIENTE ########

s1 = Singleton()
s2 = Singleton()

print(s1 == s2)         #TRUE

s1.increment()
s2.increment()

File: Singleton01.js
class Singleton {
    constructor() {
        if (!Singleton.instance) {
            Singleton.instance = this; // Se não houver uma instância Singleton, atribui a instância atual a Singleton.instance
            this.value = 0; // Define um atributo "value" na instância
        }
    
        return Singleton.instance; // Retorna a instância Singleton (se já existir) ou a instância atual (se ainda não existir)
    }
    
    increment() {
      this.value++;
      console.log(`Value: ${this.value}`);
    }
  }
  
  // CLIENTE
  
  const s1 = new Singleton(); // Cria a primeira instância de Singleton
  const s2 = new Singleton();
  const s3 = new Singleton();
  
   // Tenta criar outra instância, mas obtém a mesma instância já criada
  
  console.log(s1 === s2); // Verifica se s1 e s2 referem-se à mesma instância (true)
  
  s1.increment(); // Incrementa o valor na instância s1
  s2.increment(); // Incrementa o valor na instância s2
  s3.increment();
  